winch/codegen/src/visitor.rs

```diff
@@ -1717,6 +1717,11 @@ where
         if self.tunables.consume_fuel {
             self.emit_fuel_check();
         }
+
+        // Emit epoch check right after binding the loop header.
+        if self.tunables.epoch_interruption {
+            self.emit_epoch_check();
+        }
```

Comment:
Maybe move this `if self.tunables.epoch_interruption` into the `emit_epoch_check` helper? (And rename it to `maybe_emit_epoch_check` or something like that? Feel free to bikeshed.)

This code motion would simplify callers, which are all doing the exact same conditional.